catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -145,71 +138,88 @@ public String getId() {
+          && hasNitfResource
+          && hasOverviewDerivedImageResource
+          && hasOriginalDerivedImageResource;
+    } else {
+      return false;
     }
-    return false;
   }
 
   private boolean hasValidLocation(String location) {
-    boolean hasValidLocation = false;
-
     if (StringUtils.isNotBlank(location)) {
       try {
         // parse the WKT location to determine if it has valid format
-        WKTReader wktReader = new WKTReader(geometryFactory);
+        final WKTReader wktReader = new WKTReader(geometryFactory);
         wktReader.read(location);
-        hasValidLocation = true;
+        return true;
       } catch (ParseException e) {
-        LOGGER.debug("Location [{}] is invalid, cannot chip this image", location);
+        LOGGER.debug("Location [{}] is invalid. Cannot chip this image", location);
+      }
+    }
+
+    return false;
+  }
+
+  private static Optional<URI> getOriginalDerivedResourceUri(final Metacard metacard) {
+    final Attribute derivedResourceUriAttribute =
+        metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+    if (derivedResourceUriAttribute != null) {
+      for (Serializable value : derivedResourceUriAttribute.getValues()) {
+        if (value instanceof String) {
+          final String derivedResourceUriString = (String) value;
+          try {
+            final URI derivedResourceUri = new URI(derivedResourceUriString);
+
+            if (shouldBeChippedLocally(derivedResourceUri)) {
+              final String fragment =
+                  derivedResourceUri.getFragment(); // {@code null} if the fragment is undefined
+
+              if (StringUtils.equals(ORIGINAL_QUALIFIER, fragment)) {
+                return Optional.of(derivedResourceUri);
+              }
+            } else {
+              for (NameValuePair parameter :
+                  URLEncodedUtils.parse(derivedResourceUri, StandardCharsets.UTF_8.name())) {
+                if (QUALIFIER_KEY.equals(parameter.getName())) {
+                  final String parameterValue = parameter.getValue();
+
+                  if (StringUtils.equals(ORIGINAL_QUALIFIER, parameterValue)) {
+                    return Optional.of(derivedResourceUri);
+                  }
+                }
+              }
+            }
+          } catch (URISyntaxException e) {
+            // ignore
+          }
+        }
       }
     }
 
-    return hasValidLocation;
+    return Optional.empty();
   }
 
-  private String getQualifierForRemoteResource(String uriString) throws URISyntaxException {
-    return URLEncodedUtils.parse(new URI(uriString), StandardCharsets.UTF_8.name())
-        .stream()
-        .filter(pair -> QUALIFIER_KEY.equals(pair.getName()))
-        .map(NameValuePair::getValue)
-        .filter(ORIGINAL_QUALIFIER::equals)
-        .findFirst()
-        .orElse(""); // default
+  /**
+   * The "content" scheme in the resource.derived-uri indicates that the resource should be chipped
+   * locally.
```

Comment:
:question: Is this a firm guarantee that we will only have the resource.derived-uri be a `content:` uri if its local?

Reply:
Good question.

It is not a firm guarantee that if the scheme of the `resource.derived-uri` is `content` the resource can be chipped locally, but it's a pretty sound assumption. This method could do something fancy like inject something to check the supported schemes. I just investigated this approach, but it would not be very straightforward. This `ImagingChipActionProvider` already needs to be redesigned to support all kinds of remote sources. 

I updated this comment to clarify the assumption. I also added a comment in the `getChippingUrl` method to point out that, because of this assumption, the `Chip Image` action would not appear for a resource that can be retrieved locally that has a URI with a scheme other than `content`.

Reply:
Following up to leave a record of my thoughts on this:

I might have been abusing the word "local". Can be chipped locally, I mean that the overview and original resources can be retrieved through the local Catalog because the source supports qualifiers. This is not the case for a CSW Federated Source, which must be chipped remotely. The good news in that the `Chip Action` just won't appear in the list of Actions if we were unable to assume that it can be chipped locally.